### ISSUE TITLE
Initiate Camera on Echo Show

### ISSUE THREAD

**Issue by peterchs** [586837r/node-red-contrib-alexa-remote2#162]
I'd like to trigger my echo show to stream the camera stream when an event occurs - e.g. pir tripped or doorbell rung.

I can intiate this via voice like "alexa show front camera" but I'd like to do this triggered from node red.  Is this possible?

**Comment by craigayre** [comment]
You may be able to achieve this in a roundabout way using the new custom commands allowed at the end of routines (https://www.reddit.com/r/alexa/comments/jejjsi/alex_routine_custom_commands/)

Unfortunately this seems to only be available in the US at the moment.

You could setup a routine with a "show front door" custom command and then execute the routine through this package.

It certainly isn't ideal as it requires a routine for each specific custom command but should solve this particular use case.

Hopefully we can send custom commands through an api but this isn't something I can investigate until it's available in the UK.

**Comment by peterchs** [comment]
Sounds good - but yes in UK and this option isnt available yet, but thanks for replying will keep an eye out hopefully it'll be added to UK alexa soon.

**Comment by thecalle** [comment]
There's a Alexa Skill called Monocle RTSP/RTP IP Cameras. Using this skill you can connect every RTSP camera to Alexa Cloud. If the camera is a simple RTSP camera (missing the mandatory HTTPS certificates for Alexa connection), using the Monocle gateway, every camera can be connected.
When the camera is configured, you can say "Alexa show me 'camera_name'!" and you got the streaming on Show or video device. Alexa remote, when configuring the Home Automation devices, list it, but can't be correctly configured or actioned using NodeRed. Can you check this path and fix the camera usage?

**Comment by peterchs** [comment]
Yes I also use monocle cam and note the same problem with inability to use the cameras in a remote2 node/action thus this issue.

In the UK now the option to send custom commands has appeared in the alexa app for routines - so I can create a routine with custom command to "show front camera" and specify to run it from the show I want it to display from.   I can then initiate this routine from remote2 and it works well.   Annoyingly Alexa says 'ok' after running the routine, as she normally would when reacting to the same command spoken.  It even bumps up the volume a bit to respond if its muted :(  But anyway thanks @craigayre for flagging that forthcoming feature.

Still a workaround, so keeping issue open, hopefully remote2 can initiate the camera on the show somehow directly without using routines.

**Comment by jmorris644** [comment]
I am just getting into this. Did you find any other option to make it work? Or are you still using the routine?

**Comment by peterchs** [comment]
Using routines works well, except for the ‘ok’ response you get from the show when it’s run.

**Comment by jmorris644** [comment]
I found this "https://flows.nodered.org/node/node-red-contrib-amcrest-ad110". It works with my AD410. So now I am adding that to the strudel node (using the routine), but I am also looking into "https://flows.nodered.org/node/node-red-contrib-face-recognition" to see what I can do.
